Endava.TechCourse.BankApp.Application/Commands/AddWallet/AddWalletHandler.cs

```diff
@@ -17,17 +19,47 @@ public AddWalletHandler(ApplicationDbContext context)
 
     public async Task<CommandStatus> Handle(AddWalletCommand request, CancellationToken cancellationToken)
     {
+        var user = await context.Users
+            .Include(x => x.Wallets)
+            .FirstOrDefaultAsync(x => x.Id == request.UserId, cancellationToken);
+        if (user is null)
+            return CommandStatus.Failed($"Userul {request.UserId} nu a fost gasit");
+
+        var walletType = await context.WalletTypes
+            .FirstOrDefaultAsync(x => x.Id == request.WalletTypeId, cancellationToken);
+        if (walletType is null)
+            return CommandStatus.Failed("Tip portofel nu exista.");
+
+        var currency = await context.Currencies
+            .FirstOrDefaultAsync(x => x.Id == request.CurrencyId, cancellationToken);
+        if (currency is null)
+            return CommandStatus.Failed("Valuta nu exista.");
+
         var wallet = new Wallet
         {
-            WalletType = request.WalletType,
+            Code = Create16DigitString(),
+            WalletTypeId = request.WalletTypeId,
             Amount = request.Amount,
             CurrencyId = request.CurrencyId,
             UserId = request.UserId
         };
 
+        if (user.Wallets.Count == 0)
+            user.MainWalletId = wallet.Id;
+
         await context.Wallets.AddAsync(wallet, cancellationToken);
         await context.SaveChangesAsync(cancellationToken);
 
         return new CommandStatus();
     }
+
+    private string Create16DigitString()
```

Comment:
The method name must represent what it is doing, in your case "Generate Wallet Code"